src/components/NewWorkspaceModal.ts

```diff
@@ -319,11 +319,19 @@ const NewWorkspaceModal = withDisplayName(
     };
 
     const isBillingProjectApplicable = (project: BillingProject): boolean => {
-      // As of AJ-1164, if requireEnhancedBucketLogging is true, then azure billing projects are ineligible.
-      // This coupling of enhanced bucket logging and billing project may change in the future.
-      if (!!workflowImport || !!requireEnhancedBucketLogging) {
+      // This is used when importing data to enforce a specific cloud.
+      if (cloudPlatform && project.cloudPlatform !== cloudPlatform) {
+        return false;
+      }
+      if (workflowImport) {
         return !isAzureBillingProject(project);
       }
+      // If creating a new workspace and enhanced bucket logging is required, allow all GCP projects
+      // (user will be forced to select "Workspace will have protected data" for GCP projects)
+      // and Azure billing projects that support protected Data.
+      if (!cloneWorkspace && requireEnhancedBucketLogging && isAzureBillingProject(project)) {
```

Comment:
As far as I can tell `requireEnhancedBucketLogging` is never passed during a workspace clone operation. But nevertheless, the cloning cloud platform handling is done right below (based on the source workspace), so I limited this case to creation.